code/game/verbs/suicide.dm

```diff
@@ -31,7 +31,7 @@
 		return
 
 	if(forced || (confirm == "Yes"))
-		if(!forced && isAntag(src))
+		if(!forced && isAntag(src) && !HAS_TRAIT(src, TRAIT_RESPAWNABLE))
```

Comment:
If you're just aiming to address the ghost bar here, probably couldn't hurt to further focus the trait source? I'm not sure if this might possibly have other consequences otherwise.
```suggestion
		if(!forced && isAntag(src) && !HAS_TRAIT_FROM(src, TRAIT_RESPAWNABLE, GHOST_ROLE))
```

Reply:
After discussion on discord, this is probably fine